src/journal-remote/journal-remote-main.c

```diff
@@ -152,6 +155,24 @@ static int dispatch_http_event(sd_event_source *event,
                                uint32_t revents,
                                void *userdata);
 
+static int get_accept_encoding(char **encoding) {
+        int r;
+        float q = 1.0;
+
+        if (arg_compression.size <= 0)
+                return 0;
+
+        float step = q / arg_compression.size;
+
+        for (size_t i = 0; i < arg_compression.size; i++, q-=step) {
+                _cleanup_free_ const char *c = compression_lowercase_to_string(arg_compression.opts[i].algorithm);
+                r = strextendf_with_separator(encoding, ",", "%s;q=%.1f", TAKE_PTR(c), q);
```

Comment:
`TAKE_PTR()` is not necessary.

Please do not partially modify return value `encoding`. Please use an temporal buffer, and assign it to the return value on success.